### PR TITLE
Update resim-python-client dep to the latest version

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Also, add some code to ensure our mocks stay synced.

## Guide to reproduce test results.
```
bazel test //...
```
## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [x] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

**Review by benabineri — APPROVED** [review]

File: MODULE.bazel [hunk 1]
```diff
@@ -24,7 +24,7 @@ bazel_dep(name = "nlohmann_json", version = "3.11.3")
 bazel_dep(name = "platforms", version = "0.0.10")
 bazel_dep(name = "protobuf", repo_name = "com_google_protobuf", version = "26.0")
 bazel_dep(name = "pybind11_bazel", version = "2.12.0")
-bazel_dep(name = "resim-python-client", version = "6c485d1450e93be66cbee85f02077ff0c2b65198")
+bazel_dep(name = "resim-python-client", version = "b39d98f0e0ee9c213d5279af7f648e81e1693a92")
```
Comment (iainjwhiteside) [inline MODULE.bazel:27]:
```suggestion
bazel_dep(name = "resim-python-client", version = "185d9c00b3d7903150a3149fd7e258d9ceae12f5")
```

**Comment by iainjwhiteside** [comment]
note: there is now an even newer version

File: MODULE.bazel (hunk 1 above)
Comment (mikebauer) [reply MODULE.bazel:27]: This requires another PR first then. One sec.

Comment (mikebauer) [reply MODULE.bazel:27]: This should do it! https://github.com/resim-ai/open-core/pull/189

Comment (mikebauer) [reply MODULE.bazel:27]: Done

**Pull request merged**